content/en/user-guide/tools/transparent-endpoint-injection/dns-server.md

```diff
@@ -8,52 +8,88 @@ aliases:
+```
+
+This option disables DNS resolution of AWS domains to the LocalStack container and prevents Lambda from disabling SSL validation.
+With disabled transparent endpoint execution mode, the AWS SDK within Lambda functions might connect to the real AWS API.
+The transparent endpoint execution mode is only available in LocalStack Pro.
+
+### DNS Server bind address
+
 If you experience problems when running LocalStack and the DNS server is the issue, you can disable the DNS server using:
 
 ```bash
 DNS_ADDRESS=0
 ```
 
 {{< alert title="Warning" color="warning" >}}
-We do not recommend this configuration since this disables resolving `localhost.localstack.cloud` to the LocalStack container.
+We do not recommend disabling the DNS server since this disables resolving `localhost.localstack.cloud` to the LocalStack container.
 {{< / alert >}}
 
-You can also specify which exact URLs should be redirected to LocalStack by defining a hostname regex like:
+This option is primarily used by [LocalStack developers]({{< ref "contributing/development-environment-setup" >}}) in host mode because binding port 53 requires root privileges and port 53 might be occupied.
+
+### Fallback DNS server
+
+If you want to use another upstream DNS resolver than Google (default `8.8.8.8`),
+specify the fallback DNS server where all non-redirected queries (i.e., not matching `DNS_LOCAL_NAME_PATTERNS`) will be forwarded to:
 
 ```bash
-DNS_LOCAL_NAME_PATTERNS='.*(ecr|lambda).*.amazonaws.com'
+DNS_SERVER=1.1.1.1
 ```
 
-Using this configuration, the LocalStack DNS server only redirects ECR and Lambda domains to LocalStack, and the rest will be resolved via `$DNS_SERVER`. This can be used for hybrid setups, where certain API calls (e.g., ECR, Lambda) target LocalStack, whereas other services will target real AWS.
-
-{{< alert title="Warning" color="warning">}}
-We generally do not recommend connecting to real AWS from within LocalStack, in fact you should avoid using real AWS credentials anywhere in your LocalStack apps. Use this configuration with caution.
-{{< /alert >}}
+### Custom redirects
 
-There is the possibility to manually set the DNS server all not-redirected queries will be forwarded to:
+If you want to resolve only certain AWS URLs to LocalStack,
+specify a comma-separated list of hostname regex patterns such as:
 
 ```bash
-DNS_SERVER=1.1.1.1
+DNS_LOCAL_NAME_PATTERNS='.*(ecr|lambda).*.amazonaws.com'
 ```
 
-Per default, LocalStack uses the Google DNS resolver at `8.8.8.8`.
+Using this configuration, the LocalStack DNS server only resolves ECR and Lambda domains to LocalStack, and the rest will be resolved via the `DNS_SERVER` (i.e., the real DNS entry by default).
+This can be used for hybrid setups, where certain API calls (e.g., ECR, Lambda) target LocalStack, whereas other services will target real AWS.
+The regex pattern follows Python flavored-regex and can be tested at [regex101.com](https://regex101.com/r/OzIsQa/1).
```

Comment:
**This is why we need to rename this variable!!!!!**.

Any queries matching these regexes are _ignored_ by the LocalStack DNS server, so with TEI on anything _not_ matching these regexes is resolved by LocalStack, and `*.ecr.*.amazonaws.com` and `*.lambda.*.amazonaws.com` are forwarded to the upstream resolver.

Here we split the regex by whitespace and add each entry to the skip list in the resolver:
https://github.com/localstack/localstack/blob/4b7482d22e81bfff8c43f8b74eaeadb60bc6debf/localstack/dns/server.py#L831-L833


and here we skip local resolution if the domain matches:
https://github.com/localstack/localstack/blob/0bb7ce612be52a65c218377cdb04a14a215c6ba1/localstack/dns/server.py#L398-L409

```suggestion
Using this configuration, the LocalStack DNS server resolves all AWS domains to LocalStack _except_ ECR and Lambda domains which will be resolved via the `DNS_SERVER` (i.e., the real DNS entry by default).
This can be used for hybrid setups, where certain API calls (e.g., ECR, Lambda) target AWS, whereas other services will target LocalStack.
The regex pattern follows Python flavored-regex and can be tested at [regex101.com](https://regex101.com/r/OzIsQa/1).
```

Reply:
omg 🤦 yess 💯 
I looked up the implementation and messed it up getting confused every time I try to read the config name 🙃 

Fixed 👍